pkg/cmd/start/bootstrap.go

```diff
@@ -356,9 +356,13 @@ func setupControllers(ctx context.Context, mgr manager.Manager) {
 		os.Exit(1)
 	}
 
-	if err := appsv1controllers.NewNamespaceReconciler(client, clientReader, schema).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "Namespace")
-		os.Exit(1)
+	if viper.GetBool(v1.ConfigEnableNamespaceController) {
+		if err := appsv1controllers.NewNamespaceReconciler(client, clientReader, schema).SetupWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "Namespace")
+			os.Exit(1)
+		}
+	} else {
+		log.Trace("skipping reconciliation for namespaces, do not have permissions to list and watch namespaces")
```

Comment:
Should we print a warning instead of using trace level? what do you think?

Reply:
This log level was taken from the original code, but yeah I think a warning level is better.